### PR TITLE
[WIP] Adds sierra_bib_merger project

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Merging Sierra objects from the sierra_to_dynamo app (as provided via a DynamoDB Event Stream -> Lambda -> SNS).

In order that these objects can then be introduced to the ingestion pipeline.

This PR adds a simple skeleton for that project.

### Who is this change for?

📖 😍 

### Have the following been considered/are they needed?

- [ ] Deployed new versions
- [ ] Run `terraform apply`.

### CODE REVIEW

**Comment by kenoir** [comment]
For: https://github.com/wellcometrust/platform/issues/1196

File: project/Dependencies.scala [hunk 1]
```diff
@@ -69,7 +69,7 @@ object Dependencies {
     "org.scalatest" %% "scalatest" % versions.scalatest % "test",
     "com.fasterxml.jackson.dataformat" % "jackson-dataformat-yaml" % versions.jacksonYamlVersion % " test",
     "com.novocode" % "junit-interface" % versions.junitInterface % "test"
-  ) ++ esDependencies ++ awsDependencies ++ akkaDependencies
+  ) ++ esDependencies ++ awsDependencies ++ akkaDependencies ++ dynamoDependencies
```
Comment (alexwlchan) [inline project/Dependencies.scala]: This is a bad merge conflict.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -86,9 +86,12 @@ object Dependencies {
   val idminterDependencies
     : Seq[ModuleID] = commonDependencies ++ mysqlDependencies ++ circeDependencies
 
-  val reindexerDependencies: Seq[ModuleID] = commonDependencies ++ dynamoDependencies
+  val reindexerDependencies: Seq[ModuleID] = commonDependencies
```
Comment (alexwlchan) [inline project/Dependencies.scala]: As is this, I think.

File: sierra_adapter/sierra_bib_merger/src/main/scala/uk/ac/wellcome/platform/sierra_bib_merger/modules/SierraBibMergerModule.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import uk.ac.wellcome.metrics.MetricsSender
+import uk.ac.wellcome.platform.sierra_bib_merger.services.{SierraBibMergerUpdaterService, SierraBibMergerWorkerService}
+import uk.ac.wellcome.utils.TryBackoff
+
+object SierraBibMergerModule extends TwitterModule with TryBackoff {
+
+  override lazy val continuous: Boolean = false
+
+  val tableName: Flag[String] = flag[String](
+    name = "bibMerger.dynamo.tableName",
```
Comment (alexwlchan) [inline sierra_adapter/sierra_bib_merger/src/main/scala/uk/ac/wellcome/platform/sierra_bib_merger/modules/SierraBibMergerModule.scala]: We already have a way to build Dynamo config flags.

**Comment by alexwlchan** [comment]
Replaced by #1232.

**Pull request closed without merging**